services/common/src/components/project/ProjectDescriptionTab.tsx

```diff
@@ -0,0 +1,521 @@
+    setData([...processedEnvironmentActPermitResult]);
+    processedEnvironmentActPermitResult = [];
+  };
+
+  const loadEnvironmentActPermitData = (authorizations) => {
+    processAndSetData(
+      authorizations,
+      AMS_ENVIRONMENTAL_MANAGEMENT_ACT_TYPES,
+      ENVIRONMENTAL_MANAGMENT_ACT,
+      setEnvironmentalManagementActData
+    );
+  };
+
+  useEffect(() => {
+    loadOtherActPermitData(
+      project.project_summary.authorizations,
+      AMS_MINES_ACT_TYPE,
+      setMinesActData
+    );
+    loadOtherActPermitData(
+      project.project_summary.authorizations,
+      AMS_WATER_SUSTAINABILITY_ACT_TYPES,
+      setWaterSustainabilityActData
+    );
+    loadOtherActPermitData(
+      project.project_summary.authorizations,
+      AMS_FORESTRY_ACT_TYPE,
+      setForestryActData
+    );
+    loadEnvironmentActPermitData(project.project_summary.authorizations);
+  }, [
+    project.project_summary.authorizations,
+    transformedProjectSummaryAuthorizationTypes,
+    dropdownProjectSummaryPermitTypes,
+  ]);
+
+  const handleViewProjectDescriptionClicked = () => {
+    const url = GLOBAL_ROUTES?.EDIT_PROJECT_SUMMARY.dynamicRoute(
+      project.project_summary.project_guid,
+      project.project_summary.project_summary_guid,
+      "purpose-and-authorization",
+      false
+    );
+    history.push(url);
+  };
+
+  const transformAuthorizations = (valuesFromForm: any) => {
+    const { authorizations = {}, project_summary_guid } = valuesFromForm;
+
+    const transformAuthorization = (type, authorization) => {
+      return { ...authorization, project_summary_authorization_type: type, project_summary_guid };
+    };
+
+    let updatedAuthorizations = [];
+    let newAmsAuthorizations = [];
+    let amendAmsAuthorizations = [];
+
+    projectSummaryAuthorizationTypesArray.forEach((type) => {
+      const authsOfType = authorizations[type];
+      if (authsOfType) {
+        if (isArray(authsOfType)) {
+          const formattedAuthorizations = authsOfType.map((a) => {
+            return transformAuthorization(type, a);
+          });
+          updatedAuthorizations = updatedAuthorizations.concat(formattedAuthorizations);
+        } else {
+          newAmsAuthorizations = newAmsAuthorizations.concat(
+            authsOfType?.NEW.map((a) =>
+              transformAuthorization(type, {
+                ...a,
+                project_summary_permit_type: [AMS_AUTHORIZATION_TYPES.NEW],
+              })
+            )
+          );
+          amendAmsAuthorizations = amendAmsAuthorizations.concat(
+            authsOfType?.AMENDMENT.map((a) =>
+              transformAuthorization(type, {
+                ...a,
+                project_summary_permit_type: [AMS_AUTHORIZATION_TYPES.AMENDMENT],
+              })
+            )
+          );
+        }
+      }
+    });
+    return {
+      authorizations: updatedAuthorizations,
+      ams_authorizations: { amendments: amendAmsAuthorizations, new: newAmsAuthorizations },
```

Comment:
How similar is this function to the one in ProjectSummaryForm? If we're copy/pasting, perhaps it's time to make it a helper function.

Reply:
This is changed to a helper function now. Thanks for the help with this.